src/executor.py

```diff
@@ -17,12 +19,49 @@ def __init__(self, ast):
         for _, (_, identifier), block in ast:
             self.subroutines[identifier] = block
 
+        self.operations = {
+            "add": self.instruction_add,
+            "subtract": self.instruction_subtract,
+            "multiply": self.instruction_multiply,
+            "divide": self.instruction_divide,
+            "and": self.instruction_and,
+            "or": self.instruction_or,
+            "xor": self.instruction_xor,
+            "negate": self.instruction_negate,
+            "not": self.instruction_not,
+            "if": self.instruction_if,
+            "while": self.instruction_while,
+            "exit": lambda args, bindings: exit(self.resolve_value(args[0], bindings)),
+            "print": lambda args, bindings: print(self.resolve_value(args[0], bindings)),
+            "push": self.instruction_push,
+            "pop": self.instruction_pop,
+            "invocation": self.instruction_invocation,
+            "stop": lambda: True,
+            "store": self.instruction_store,
+            "load": self.instruction_load,
+        }
+
     def execute_block(self, block, bindings: Bindings):
         for instruction in block:
             stop = self.execute_instruction(instruction, bindings)
             if stop:
                 return True
 
+    def execute_instruction(self, instruction, bindings: Bindings):
+        operation, *args = instruction
+
+        if operation not in self.operations:
+            print("Error: invalid operation", operation)
+            return False
```

Comment:
This isn't good error handling. This should blow up. As it is it doesn't even interrupt program execution.